src/otaclient_common/linux.py

```diff
@@ -302,3 +303,34 @@ def fstrim_at_subprocess(
         stderr=subprocess.DEVNULL,
         shell=True,
     )
+
+
+def is_regular_file(path: Path) -> bool:
+    """Check if a file is regular."""
+    try:
+        path_stat = path.lstat()
+        return (
+            stat.S_ISREG(path_stat.st_mode)
+            and not stat.S_ISLNK(path_stat.st_mode)
+            and path_stat.st_size > 0
+        )
+    except OSError:
+        return False
```

Comment:
Suggest to rename it to `is_non_empty_regular_file` to align with the implementation of this function.